### ISSUE TITLE
[Feature]: Azure, OpenAI - Log Response Headers 

### ISSUE THREAD

**Issue by ishaan-jaff** [BerriAI/litellm#1724]
### The Feature

User request 

### Motivation, pitch

cc @Manouchehri

**Comment by Manouchehri** [comment]
Probably wouldn't hurt to do for other providers as well. Off the top of my head, I don't _think_ any popular LLM API is returning private/sensitive content in their response headers.

**Comment by ishaan-jaff** [comment]
Starting with just OpenAI/Azure because we're going to need to move all OpenAI python calls 

from `chat.completions.create`
to `chat.completions.with_raw_response.create`

**Comment by ishaan-jaff** [comment]
Tradeoffs of this move: https://github.com/openai/openai-python/issues/416#issuecomment-1919830366 It looks like `with_raw_response` will currently return an object with only sync methods

**Comment by ishaan-jaff** [comment]
another approach is using: https://www.python-httpx.org/advanced/#event-hooks

**Comment by ishaan-jaff** [comment]
Useful details here: https://github.com/langchain-ai/langchain/issues/9601

**Comment by Manouchehri** [comment]
Feedback from Langfuse's Marc (when I asked if he thought it was a good idea to also feed upstream traces to a different project):

> I'd suggest to add it to the metadata on the same generation in Langfuse though. Splitting the data into two projects makes this a kind of "exotic" setting that not many users will benefit from

I'm inclined to agree with him, not a good idea if it's just me asking.

For logging the request and response metadata, can we do it like this?

https://github.com/orgs/langfuse/discussions/1070#discussioncomment-8369918

```json
{
  "request": {
    "headers": {
      "host": "example.com"
    },
    "url": "https://api.openai.com/v1/chat/completions"
  },
  "response": {
    "headers": {
      "age": "1337"
    },
    "status": 200
  }
}
```

**Comment by ishaan-jaff** [comment]
Attempt 1 PR: https://github.com/BerriAI/litellm/pull/1873 
TLDR: we can't use `chat.completions.with_raw_response.create` it's not async and impacts throughput

**Comment by Manouchehri** [comment]
Hmm, I noticed that Sentry is able to log the response object/headers. Maybe we can steal/borrow their method of doing it?

**Comment by Manouchehri** [comment]
This is possible to do with: https://til.simonwillison.net/httpx/openai-log-requests-responses

**Comment by ishaan-jaff** [comment]
we do this now
